primer.md

```diff
@@ -238,23 +238,54 @@ there are errors during the processing of a CloudEvent, the software
 reporting to report them.
 
-## Versioning of Attributes
-
-For certain CloudEvents attributes, the entity or data model referenced by its
-value might change over time. For example, `dataschema` might reference one
-particular version of a schema document. Often these attribute values will then
-distinguish each variant by including some version-specific string as part of
-its value. For example, a version number (`v1`, `v2`), or a date (`2018-01-01`)
-might be used.
-
-The CloudEvents specification does not mandate any particular pattern to be
-used, or even the use of version strings at all. This decision is up to each
-event producer. However, when a version-specific string is included, care should
-be taken whenever its value changes as event consumers might be reliant on the
-existing value and thus a change could be interpreted as a "breaking change".
-Some form of communication between producers and consumers should be established
-to ensure the event consumers know what possible values might be used. In
-general, this is true for all CloudEvents attributes as well.
+## Versioning of CloudEvents
+
+For many CloudEvents, the schema of the data within the event might
+change over time. The CloudEvents specification does not mandate any
+particular pattern to be used, or even the use or consideration of
+versioning at all. This decision is up to each event producer.
+
+However, event producers are encouraged to consider how they might
+evolve their schema without breaking consumers. Two specific
+context attributes are particularly important in this respect, but
+differ in expected usage: `type` and `dataschema`.
+
+The `type` attribute is expected to be the primary means by which
+consumers identify the type of event that they receive. This could
+be achieved by subscribing to a specific CloudEvent type, or by
+filtering all received CloudEvents by type locally. But consumers
+who have identified a CloudEvent type will generally expect the data
+within that type to only change in backwardly-compatible ways,
+unless clearly indicated otherwise. The precise meaning of
+"backwardly-compatible" will vary by data content type.
+
+The `dataschema` attribute is expected to be informational, largely
+to be used during development and by tooling that is able to provide
+diagnostic information over arbitrary CloudEvents with a data
+content type understood by that tooling.
+
+When a CloudEvent's data changes in a backwardly-compatible way, the
+value of the `type` attribute should generally stay the same. The
+value of the `dataschema` attribute may change, or it may stay the
+same but the content served from the URI can change to reflect the
+updated schema.
+
+When a CloudEvent's data changes in a backwardly-incompatible way,
+the value of the `type` attribute should generally change. The event
+producer is encouraged to produce both the old event and the new
+event for some time (potentially forever) in order to avoid
+disrupting consumers.
```

Comment:
nit: this paragraph of content seems to belong in the paragraph starting at 253 where the first half discussing changes both breaking and non-breaking was given.  Putting them together may increase readability.

Reply:
I've moved things around quite a bit now... as you suggested, it was a matter of "top-level, then more detail" but it's now reorganized to treat each attribute separately. That's also involved moving the general encouragement to earlier on, which I'm generally pleased about. See what you think.